elki/src/main/java/ch/ethz/globis/pht/util/PhIteratorBase.java

```diff
@@ -0,0 +1,19 @@
+/*
+ * Copyright 2011-2014 ETH Zurich. All Rights Reserved.
+ *
+ * This software is the proprietary information of ETH Zurich.
+ * Use is subject to license terms.
+ */
+package ch.ethz.globis.pht.util;
+
+import java.util.Iterator;
+
+public interface PhIteratorBase<K, V, E> extends Iterator<V> {
```

Comment:
This is the reason we don't use the Java `Iterator` API. Is the `extends Iterator<V>` sensible? This could also implement `Iterator<K>` and `Iterator<E>`, so it feels a bit dangerous to overload it this way.
Just a comment, no need to change this.